source/slang/slang-emit-glsl.cpp

```diff
@@ -2645,33 +2652,45 @@ void GLSLSourceEmitter::emitVarDecorationsImpl(IRInst* varDecl)
     for (auto decoration : varDecl->getDecorations())
```

Comment:
the function was mostly used for raytracing, but it does have another modifier:  `IRGloballyCoherentDecoration`.

it may need to be seperated, but for now it is the only place which makes sense since it is an optional modifier.

About the comment, I don't know why it is there?

Reply:
What I suggested is to not change the existing for-loop.
And add one more loop that does a similar thing.
```
for (auto decoration : varDecl->getDecorations())
{
... keep ray tracing stuff as it was ...
}
for (auto decoration : varDecl->getDecorations())
{
... add your new stuff here ...
}
```

Reply:
that sounds good, will do